vigranumpy/test/test_segmentation.py

```diff
@@ -33,6 +33,49 @@ def test_labelMultiArray():
     _impl_test_labelMultiArray(numpy.uint8)
     _impl_test_labelMultiArray(numpy.uint32)
     _impl_test_labelMultiArray(numpy.float32)
+
+
+def _impl_test_applyMapping(dtype):
+    original = numpy.arange(100, dtype=dtype ).reshape(10,10)
+    mapping = { int(k) : int(k + 100) for k in original.flat[:] }
+
```

Comment:
I don't know of a general solution, but in case of the labelmap, you could use the promote type:
```
typedef typename NumericTraits<SrcVoxelType>::Promote KeyType;
typedef typename NumericTraits<DestVoxelType>::Promote ValueType;
typedef boost::unordered_map<KeyType, ValueType> labelmap_t;
labelmap[extract<KeyType>(key)] = extract<ValueType>(value);
```